cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -392,9 +394,17 @@ func (s *Supervisor) startOpAMP() error {
 func (s *Supervisor) startOpAMPClient() error {
 	s.opampClient = client.NewWebSocket(newLoggerFromZap(s.logger))
 
-	tlsConfig, err := s.config.Server.TLSSetting.LoadTLSConfig(context.Background())
+	// determine if we need to load a TLS config or not
+	var tlsConfig *tls.Config
+	url, err := url.Parse(s.config.Server.Endpoint)
```

Comment:
nit: to avoid colliding with the `url` package name
```suggestion
	parsedUrl, err := url.Parse(s.config.Server.Endpoint)
```